pandas/core/dtypes/missing.py

```diff
@@ -63,7 +78,29 @@
 _dtype_str = np.dtype(str)
 
 
-def isna(obj):
+@overload
+def isna(obj: Scalar) -> bool:
+    ...
+
+
+@overload
+def isna(
+    obj: IndexOpsMixin | ExtensionArray | np.ndarray | list,
+) -> npt.NDArray[np.bool_]:
+    ...
+
+
+@overload
+def isna(obj: NDFrameT) -> NDFrameT:
+    ...
+
+
+@overload
+def isna(obj: object) -> bool | npt.NDArray[np.bool_] | NDFrame:
+    ...
+
+
+def isna(obj: object) -> bool | npt.NDArray[np.bool_] | NDFrame:
```

Comment:
should this be `obj: Any` instead of `obj: object`?

Reply:
typeshed [recommends](https://github.com/python/typeshed/blob/master/CONTRIBUTING.md) using object when literally anything is accepted:

> Note that Any is not the correct type to use if you want to indicate that some function can accept literally anything: in those cases use object instead.